internal/test-framework/cmd/basic_integration_run.go

```diff
@@ -0,0 +1,209 @@
+		// nolint: errcheck
+		defer os.RemoveAll(talosConfig)
+	}
+
+	for _, check := range retryChecks(ctx) {
+		if err = cRunner.Check(ctx, check); err != nil {
+			log.Fatal(err)
+		}
+	}
+
+	for _, check := range oneShotChecks(ctx) {
+		if err = cRunner.Run(ctx, check); err != nil {
+			log.Fatal(err)
+		}
+	}
+
+	return nil
+}
+
+func createCluster(ctx context.Context, osctl string, cRunner runner.Runner) (err error) {
+	// Skip basic integration cluster creation if it already exists
+	// and we have a valid talosconfig
+	if clusterExists(ctx, cRunner) {
+		_, err = os.Stat(talosConfig)
+		if os.IsNotExist(err) {
+			return fmt.Errorf("talosconfig %q does not exist but %s cluster already running", talosConfig, clusterName)
+		}
+
+		return nil
+	}
+
+	runner, err := localRunner.New()
+	if err != nil {
+		return err
+	}
+
+	args := []string{
+		"--talosconfig",
+		talosConfig,
+		"cluster",
+		"create",
+		"--name",
+		clusterName,
+		"--masters",
+		"3",
+		"--mtu",
+		"1440",
+		"--cpus",
+		"4.0",
+	}
+
+	if talosImage != "" {
+		args = append(args, "--image", talosImage)
+	}
+
+	clusterCreate := checker.Check{
+		Command: exec.CommandContext(ctx, osctl, args...),
+		Name:    "Creating cluster",
+	}
+
+	if err = runner.Run(ctx, clusterCreate); err != nil {
+		return err
+	}
+
+	args = []string{
+		"--talosconfig",
+		talosConfig,
+		"config",
+		"endpoint",
+		"10.5.0.2",
+	}
+
+	osctlEndpoint := checker.Check{
+		Command: exec.CommandContext(ctx, osctl, args...),
+		Name:    "Set osctl endpoint",
+	}
+
+	if err = runner.Run(ctx, osctlEndpoint); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func createRunnerContainer(osctl string, iTest string) (runner.Runner, error) {
+	var err error
+
+	runner, err := containerRunner.New(
+		containerRunner.WithImage(runnerImage),
+		containerRunner.WithClient(client.FromEnv),
+		containerRunner.WithTTY(true),
+		containerRunner.WithLabel("talos.owned", "true"),
+		containerRunner.WithLabel("talos.integration.name", clusterName),
+		containerRunner.WithClusterName(clusterName),
+		containerRunner.WithMount(mount.Mount{
+			Type:   mount.TypeBind,
+			Source: osctl,
+			Target: "/bin/osctl",
+		},
+		),
+		containerRunner.WithMount(mount.Mount{
+			Type:   mount.TypeBind,
+			Source: iTest,
+			Target: "/bin/integration-test",
+		},
+		),
+		containerRunner.WithMount(mount.Mount{
+			Type:   mount.TypeBind,
+			Source: filepath.Dir(talosConfig),
+			Target: filepath.Dir(talosConfig),
+		},
+		),
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	return runner, err
+}
+
+func clusterExists(ctx context.Context, cRunner runner.Runner) bool {
+	filters := filters.NewArgs()
+	filters.Add("label", fmt.Sprintf("%s=%s", "talos.owned", "true"))
+	filters.Add("label", fmt.Sprintf("%s=%s", "talos.cluster.name", clusterName))
+
+	containers, err := cRunner.(*container.ContainerRunner).Client.ContainerList(ctx, types.ContainerListOptions{Filters: filters})
+	if err != nil {
+		log.Fatal(err)
```

Comment:
idk if we want to return error back, but I imagine having some context here is going to help a lot, otherwise whole process crashes with some error like "context canceled", and we have zero idea on the context of the failure